### ISSUE TITLE
add bytes assert

### ISSUE THREAD

**Issue by lepapareil** [Orange-OpenSource/hurl#162]
In general, existing http clients (like curl) allow to retrieve the following informations : 

- status : *assert already exists*
- header : *assert already exists*
- cookie : *assert already exists*
- body : *assert already exists*
- duration : *assert already exists*
- bytes : ~assert already exists~

It would be interesting if we could test the total amount of bytes received:

```
GET https://my_api.com

[Asserts]
bytes lessThan 322
```

What do you think about it ?

**Comment by fabricereix** [comment]
that could indeed useful.
You mean the response body bytes, not the full http response bytes?

'bytes' should represent more the bytes, not the number of bytes
your assert should probably be more like:
```
[Asserts]
bytes countLessThan 322
```

**Comment by lepapareil** [comment]
Yes the response body bytes.

**Comment by jcamiel** [comment]
Interesting, and related to discussion around bytes literal in #102.

`body` query (either explicit or implicit, in request and response) has already
a text and byte semantic meaning in Hurl File.

Implicit body assert with text semantic:

```
GET https://dummy.com
HTTP/*
{"selected": true}
```

Implicit body assert with byte semantic:

```
GET https://dummy.com
HTTP/*
file,data.bin;
```

```
GET https://dummy.com
HTTP/*
hex,56FF906600AE;
```

With explicit asserts, body query is unambigously refering to bytes or text.

Explicit body assert with text semantic:

```
GET https://dummy.com
HTTP/*
body startsWith "something"
```

Explicit body assert with byte semantic (for instance, checking if there is a BOM):

```
GET https://dummy.com
HTTP/*
body startsWith hex,EFBBBF;
```


So `body` query is already representing either decoded body text, or raw byte of the request 
and body response. With a `byte` query, this two asserts will check the same thing:

```
GET https://dummy.com
HTTP/*
body equals file,data.bin;
```

```
GET https://dummy.com
HTTP/*
bytes equals file,data.bin;
```

So currently `body` is already a bytes query. On the other hand, I agree that actually `countEquals` is ambigous with a `body` query :

```
GET https://dummy.com
HTTP/*
body countEquals 1234;
```

Does this assert checks the number of raw bytes of the body response, of does it check the number of "character" of the decoded body text response. In this case, I tend to see the body as a byte collection and prefer it to test the byte size.

**Comment by fabricereix** [comment]
bytes query has been implemented in #203 
it returns all the bytes (not the count)

In order to test the number of bytes, the `countEquals` predicate can be used
```bytes countEquals 10```
